src/d3a/models/strategy/load_hours.py

```diff
@@ -291,17 +306,28 @@ def event_tick(self):
         self.bid_update.increment_update_counter_all_markets(self)
 
     def event_offer(self, *, market_id, offer):
+        """Automatically react to offers in single-sided markets.
+
+        This method is triggered by the OFFER event.
+        """
```

Comment:
Here too. You would need to add a docstring vor all `event_offer` methods in the codebase to be consistent.
But there are a lot. So you can leave this here. But the coder has to know the architecture of the d3a anyway where he/she would know what this function does and how it gets triggered. 
Dont want to start a discussion and dont expect a pitch for docstrings from you here, just want to mention . :-)

Reply:
In this case I don't completely agree with you, even though I understand your point. 
This method slowed me down a bit, as I could not immediately understand what it was doing, so I was following method calls, constant values and wiki articles. Some method names have the `event_` prefix even though they don't have an event trigger, which is sometimes confusing. 
I think that the coder often gains knowledge about the architecture by the code, and not just the opposite (especially for public repos). That's why I added this docstring in the end, hoping that when we will onboard new people it will help a bit :)

Reply:
Ok, convinced. But to be consistent, you would need to add this docstring to all `event_offer`, `event_activate`, `event_bid_traded`, `event_market_cycle` etc. methods, right?
Otherwise the uneducated user would have to find this docstring by accident. Or how would you propose to handle it?

Reply:
Ideally yes, I would add it to all those methods. In practice, this current PR is growing together with the implementation of the Home Meter device so I'm refactoring things as I go along.
We could either keep this refactoring "incomplete", or I can complete it now for all those methods. I would personally go for the second option, since this is a specific PR aimed at refactoring. If the refactoring was made inside a feature PR, maybe this would be less relevant.

Reply:
For the future, I'm thinking that even these refactoring PRs should be smaller so that reviewing them could be almost matter of seconds.